src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -263,10 +265,30 @@ public DatabaseTarget(string name) : this()
         public CommandType CommandType { get; set; }
 
         /// <summary>
+        /// Gets or sets property name of the SQL command parameter to set parameter DbType.
+        /// </summary>
+        /// <remarks>
+        /// May set strong DbType, for SQL Server is SqlDbType.
+        /// </remarks>
+        /// <docgen category='SQL Statement' order='12' />
+        [DefaultValue("DbType")]
+        public string ParameterDbTypePropertyName { get; set; }
+
+        /// <summary>
+        /// Gets or sets converter type of the SQL command parameter value.
+        /// </summary>
+        /// <docgen category='SQL Statement' order='13' />
+        [DefaultValue(typeof(DatabaseParameterConverter))]
+        public Type ParameterConverterType { get; set; }
```

Comment:
this class is not sealed, so I think subclassing is a better option here (with a virtual method). Do you agree?

Reply:
Yes, class 'DatabaseParameterConverter' only implement general methods, others can override this class to support more DbType..